### ISSUE TITLE
General feedback

### ISSUE THREAD

**Issue by serkanatpaystack** [ooade/react-starwars#5]
I like the way you have separated your components, but I would have been happier if I saw:

* components be more generic. for instance, dropdown requires the movies array and each movie object to be in a specific structure. a cleaner separation of concerns would have resulted in a dropdown that only requires an array of label and value props
* index.js is too bloated, it would have better been kept as the renderer
* network calls could have been separated into a utility method (eg `fetchMovies()`), therefore making it easier to read the code in the react lifecycle. Seeing all the network and sorting logic in there convolutes the code.
* there's no network error handler. what happens if fetch throws?
* use of `String` constructor feels unnatura
* characters encapsulate a whole new set of network logic and state, is there a better way to support ui from logic logic from api calls perhaps?
* why reduce in fetchAllCharacters? It looks like you're doing a map actually.
* render function of the character list contains lots of logic. also, each render creates a new instance of the sort function, thereby degrading applicatin memory profile

These are some of the issues that stand out among the rest.

All in all, I like what I am seeing and it feels to me like you have potential.

You can leave all of these, need not fix any of them except #2 for which I'd like you to create a new branch, fix it in that branch and then create a PR. Please then assign me and @olumytee as reviewers to that PR

Thank you.

**Comment by ooade** [comment]
Yeah, the reduce could have been a map instead :man_facepalming: 
Okay, thanks for the general feedback.

**Comment by serkanatpaystack** [comment]
By the way, this is meant to be constructive, seeing a big list of "issues" should not intimidate you, these are merely observations. You may also disagree with some of them.

The list is provided here as a gateway for you into the things we care about or have preferences over :)

**Comment by ooade** [comment]
You're right about all of them. I could have done them better. Thanks for the feedback.
